hilti/include/rt/type-info.h

```diff
@@ -0,0 +1,1071 @@
+// Copyright (c) 2020 by the Zeek Project. See LICENSE for details.
+
+#pragma once
+
+#include <optional>
+#include <string>
+#include <variant>
+#include <vector>
+
+#include <hilti/rt/exception.h>
+#include <hilti/rt/fmt.h>
+#include <hilti/rt/types/all.h>
+#include <hilti/rt/types/bytes.h>
+#include <hilti/rt/types/stream.h>
+
+namespace hilti::rt {
+
+struct TypeInfo;
+
+namespace type_info {
```

Comment:
Couple general issues in this file:

- We usually require rt types to be default constructable so they can e.g., be used in global variables. Most of these classes do not have default constructors. Feel free to drop this if you think this is not an issue.
- Many classes in this file have `const` member variables so they are e.g., not assignable. Unless this is deliberate I'd suggest we drop the `const` qualifiers.

Reply:
The type info instances are generated exclusively as `static ...`` globals, and should never change. That's why the members are `const` (maybe it helps the compiler with optimizing something). Similar reasoning re/  the default ctors: They won't be used at all, and it may in fact be better to not have them so that they don't accidentally get used.